app/scripts/metamask-controller.actions.test.js

```diff
@@ -157,67 +157,49 @@ describe('MetaMaskController', function () {
   });
 
   describe('#importAccountWithStrategy', function () {
-    it('two sequential calls with same strategy give same result', async function () {
-      let keyringControllerState1;
-      let keyringControllerState2;
+    it('throws an error when importing the same account twice', async function () {
       const importPrivkey =
         '4cfd3e90fc78b0f86bf7524722150bb8da9c60cd532564d7ff43f5716514f553';
-
       await metamaskController.createNewVaultAndKeychain('test@123');
-      await Promise.all([
-        metamaskController.importAccountWithStrategy('privateKey', [
-          importPrivkey,
-        ]),
-        Promise.resolve(1).then(() => {
-          keyringControllerState1 = JSON.stringify(
-            metamaskController.keyringController.state,
-          );
-          metamaskController.importAccountWithStrategy('privateKey', [
-            importPrivkey,
-          ]);
-        }),
-        Promise.resolve(2).then(() => {
-          keyringControllerState2 = JSON.stringify(
-            metamaskController.keyringController.state,
-          );
-        }),
+
+      await metamaskController.importAccountWithStrategy('privateKey', [
+        importPrivkey,
       ]);
-      assert.deepEqual(keyringControllerState1, keyringControllerState2);
+
+      await assert.rejects(async () => {
+        await metamaskController.importAccountWithStrategy('privateKey', [
+          importPrivkey,
+        ]);
+      }, 'Error: KeyringController - The account you are trying to import is a duplicate');
     });
   });
 
   describe('#createNewVaultAndRestore', function () {
     it('two successive calls with same inputs give same result', async function () {
-      const result1 = await metamaskController.createNewVaultAndRestore(
-        'test@123',
-        TEST_SEED,
-      );
-      const result2 = await metamaskController.createNewVaultAndRestore(
-        'test@123',
-        TEST_SEED,
-      );
+      await metamaskController.createNewVaultAndRestore('test@123', TEST_SEED);
+      const result1 = metamaskController.keyringController.state;
+      await metamaskController.createNewVaultAndRestore('test@123', TEST_SEED);
+      const result2 = metamaskController.keyringController.state;
```

Comment:
Aren't `result1` and `result2` references? We should do a deep copy here no?

Reply:
BaseController's `state` property is a getter so it will be equal to a function call returning a different object everytime `.state` is read

If we try to add `await metamaskController.keyringController.addNewAccount()` before the second `.state` call the test will fail

Reply:
thanks for clarifying! :)